src/main/java/org/folio/circulation/domain/LoanRepresentation.java

```diff
@@ -82,17 +81,17 @@ private JsonObject extendedLoan(JsonObject loan, Item item) {
     return loan;
   }
 
-  private void additionalAccountProperties(JsonObject loanRepresentation, Collection<Account> accounts) {
-    if (accounts == null) {
+  private void additionalAccountProperties(JsonObject loanRepresentation, Loan loan) {
+    if (loan.getAccounts() == null) {
       return;
     }
-    double remainingFeesFines = accounts.stream().filter(Account::isOpen)
-      .map(Account::getRemainingFeeFineAmount).reduce(Double::sum).orElse(0d);
 
     JsonObject feesAndFinesSummary = loanRepresentation.containsKey(LoanProperties.FEESANDFINES)
       ? loanRepresentation.getJsonObject(LoanProperties.FEESANDFINES)
       : new JsonObject();
-    write(feesAndFinesSummary, "amountRemainingToPay", remainingFeesFines);
+
+    write(feesAndFinesSummary, "amountRemainingToPay", loan.getRemainingFeeFineAmount()
+      .toDouble());
```

Comment:
Why does the loan representation include the amount of fees remaining to pay?

Does this mean this is calculated for every loan that is fetched in the system?

Reply:
I'm not sure. This logic was before, and i just refactored it to use new `FeeAmount` class.

> Does this mean this is calculated for every loan that is fetched in the system?

Probably not, accounts for loan have to be fetched separately and they should not be fetched each time we fetch loan.

Reply:
Ok